pkg/cloud/amazon/s3_storage.go

```diff
@@ -517,6 +517,7 @@ func newClient(
 	if conf.endpoint != "" {
 		opts.Config.Endpoint = aws.String(conf.endpoint)
 		opts.Config.S3ForcePathStyle = aws.Bool(true)
+		opts.Config.S3DisableContentMD5Validation = aws.Bool(true)
```

Comment:
Is this where you intended to set this?  I _think_ this will only apply to S3 urls that have set a custom endpoint?

Reply:
Whoops, no I meant to put it in an `else` of this block

Reply:
Done.